ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java

```diff
@@ -1915,15 +1915,26 @@ private static String getFullyQualifiedName(String... names) {
     return result;
   }
 
-  public static long getAvailableMemory(Configuration conf) {
-    int memory = HiveConf.getIntVar(conf, HiveConf.ConfVars.HIVETEZCONTAINERSIZE);
-    if (memory <= 0) {
-      memory = conf.getInt(MRJobConfig.MAP_MEMORY_MB, MRJobConfig.DEFAULT_MAP_MEMORY_MB);
-      if (memory <= 0) {
-        memory = 1024;
+  /**
+   * Get the Container Memory Size from given conf (in MB).
+   * Returns HIVETEZCONTAINERSIZE when set, otherwise falls back to MAP_MEMORY_MB.
+   * When MAP_MEMORY_MB is explicitly set to "-1" uses DEFAULT_MAP_MEMORY_MB (1024) to avoid failures.
+   * @param conf Configuration
+   * @param isTez true if in Tez mode
+   * @return Container Memory Size in MB
+   */
+  public static int getAvailableMemory(Configuration conf, boolean isTez) {
+    int containerMemSizeMb = HiveConf.getIntVar(conf, HiveConf.ConfVars.HIVETEZCONTAINERSIZE);
```

Comment:
Not sure why `isTez` was added here,... why return anything if it's not Tez as this method is strictly for determining Tez container size.

Reply:
The reason is I reused the logic for MR mode: https://github.com/apache/hive/pull/1933/files#diff-e3956e96fab0a8e9604b0e484a2ee7db29b83a62fdae4d08de068e4712191663L67